sdk/spring-sdk-testkit/src/test/java/springsdk/testmodels/eventsourced/CounterEventSourcedEntity.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package springsdk.testmodels.eventsourced;
+
+import kalix.javasdk.eventsourcedentity.EventSourcedEntity;
+import kalix.springsdk.annotations.EventHandler;
+
+import java.util.List;
+
+public class CounterEventSourcedEntity extends EventSourcedEntity<Integer> {
+
+  public Effect<String> increaseBy(Integer value) {
+    if (value <= 0) return effects().error("Can't increase with a negative value");
+    else return effects().emitEvent(new Increased(value)).thenReply(__ -> "Ok");
+  }
+
+  public Effect<String> doubleIncreaseBy(Integer value) {
+    if (value < 0) return effects().error("Can't increase with a negative value");
+    else {
+      Increased event = new Increased(value);
+      return effects().emitEvents(List.of(event, event)).thenReply(__ -> "Ok");
+    }
+  }
```

Comment:
This test is important. We must ensure that the test infra is set the current state after applying each event. 

That was not the case before, because the gRPC generated code has always the state as parameter of the event handler. For the Spring SDK, users are expected to use call `currentState` inside each event handler. As such, the `EventSourcedEntityEffectsRunner` must take that info account and we want to have this test to prove that we are doing the right thing.

Reply:
nice catch 👍